File: C++/0051. 数字金字塔.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
  
    int n;
    while(cin>>n){
        for(int i=1;i<=n;++i){
            string s(n-i,' ');
            for(int j=1;j<=i;++j){
                s = s + char('0'+j);
            }
            for(int j=i-1;j>0;--j){
                s = s + char('0'+j);
            }
            cout<<s<<endl;
        }      
    }
    
    return 0;
}

File: Python/0051. 数字金字塔.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    
    while True:
        try:
            n = int(input())
            
            for i in range(1,n+1):
                print(' '*(n-i)+('').join(list(str(i) for i in range(1,i+1)))+('').join(list(str(i) for i in range(i-1,0,-1))))
            
        except EOFError:
            break

    
if __name__ == "__main__":
    func()
